tests/drivers/gpio/gpio_basic_api/boards/native_posix.overlay

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	resources {
+		compatible = "test,gpio_basic_api";
```

Comment:
Compatible strings should use - not _

Reply:
Created #29689. That's a change that should be applied to the actual dts binding as well as all existing boards.